packages/transaction-controller/src/TransactionController.ts

```diff
@@ -695,8 +756,13 @@ export class TransactionController extends BaseController<
       transactionMeta.status = TransactionStatus.submitted;
       this.updateTransaction(transactionMeta);
       this.hub.emit(`${transactionMeta.id}:finished`, transactionMeta);
+      // must set transaction to submitted/failed before releasing lock
+      nonceLock.releaseLock();
```

Comment:
Nit: May as well move this to the `finally` block to avoid repeating it twice. It should be equivalent, so long as it's before `releaseLock()` is called.